src/Symfony/Component/HttpClient/MockHttpClient.php

```diff
@@ -39,7 +39,7 @@ public function __construct($responseFactory = null, string $baseUri = null)
             $responseFactory = [$responseFactory];
         }
 
-        if (null !== $responseFactory && !\is_callable($responseFactory) && !$responseFactory instanceof \Iterator) {
+        if (!$responseFactory instanceof \Iterator && null !== $responseFactory && !\is_callable($responseFactory)) {
```

Comment:
Same here

Reply:
Same reason as in https://github.com/symfony/symfony/pull/31018#discussion_r273341842